core/src/main/java/gyro/core/validation/CollectionMaxValidator.java

```diff
@@ -0,0 +1,22 @@
+package gyro.core.validation;
+
+import java.util.Collection;
+
+import gyro.core.resource.Diffable;
+
+public class CollectionMaxValidator implements Validator<CollectionMax> {
+
+    @Override
+    public boolean isValid(Diffable diffable, CollectionMax annotation, Object value) {
+        if (value instanceof Collection) {
+            return ((Collection<?>) value).size() <= annotation.value();
+        }
+
+        return true;
```

Comment:
If it doesn't meet the condition, then return `false`.

Reply:
The validator is for anything of collection type, so if used on a non collection type, like Integer and such it should not throw an error, and that is why by default it returns true if its not a collection, but validates otherwise.

Reply:
As discussed offline, this is more about consistency. Let's revisit this once we have a concrete decision.